docs/source/essentials/server.md

```diff
@@ -159,11 +95,55 @@ And the server should return a simple response:
 ```json
 {
   "data": {
-    "hello": null
+    "hello": "world"
   }
 }
 ```
 
+Your server itself is hosted at http://localhost:4000/graphql. This would be the endpoint you pass to set up Apollo Client.
+
+<h2 id="integrations">Server integrations</h2>
+
+Depending on whether we are creating a new application or an existing application, the method of importing `apollo-server` will vary slightly since Apollo Server must adapt to the semantics of existing servers (e.g. Express, Hapi, etc.)
+
+Both import methods will use the `apollo-server` module we installed in the previous step, but existing applications will append the desired middleware as a path-based import (e.g. `apollo-server/<variant>`, where `<variant>` would be `express`, `koa`, etc.).
+
+<h3 id="middleware">Middleware</h3>
+
+Existing applications generally already have existing middleware in place and Apollo Server works with those existing middleware options.  But in order to play along, we need to import the `ApolloServer` which corresponds to the type of server which is already in use.
+
+For example, if the application is already using an Express server, it is important to use the `apollo-server/express` import.
+
+It will also be necessary to access the instance of the existing "app" to add the new Apollo Server middleware.  Be sure to add this new import where your existing application is available.
+
+For existing applications, we'll also pass the existing application into the constructor as `app`.  New applications should not pass `app`.
+
+> The existing application is frequently already named `app`, especially when using Express.  If the application is identified by a different variable, pass the existing app in as `app`.
+
+```js
+const { ApolloServer, gql } = require('apollo-server/express');
+const { typeDefs, resolvers } = require('./schema');
+
+const server = new ApolloServer({
+  // Important: Only set `app` for existing applications!
+  app: app,
+
+  // These will be defined for both new or existing servers
+  typeDefs,
+  resolvers,
+});
+```
+
+When adding Apollo Server to an existing server, it's necessary to indicate, relative where to activate the middleware relative to other middlewares already in the app.  To do this, add the following line along with your other middleware, preferably before any of the others:
+
+```js
+server.applyMiddleware();
+```
+
+<h3 id="serverless">Serverless</h3>
+
+Apollo Server works great in so-called "serverless" environments such as Amazon Lambda and Microsoft Azure Functions.  These implementations have some extra considerations which won't be covered in this guide, but more information is provided in [Server-less]().
```

Comment:
Do we already have an idea of where `Server-less` will be linked to?

Reply:
no idea yet, this section needs work

Reply:
Maybe we should just have them open an issue if that's what they're interested in so we can gauge interest and let them know when our work on this is more ready?